tap_shopify/context.py

```diff
@@ -19,3 +22,20 @@ def is_selected(cls, stream_name):
         stream = cls.get_catalog_entry(stream_name)
         stream_metadata = metadata.to_map(stream['metadata'])
         return metadata.get(stream_metadata, (), 'selected')
+
+    @classmethod
+    def get_results_per_page(cls, default_results_per_page):
+        results_per_page = default_results_per_page
```

Comment:
Is this line just so that you can use `results_per_page` in the try block?

`get_results_per_page()` never returns `default_results_per_page` right?

Reply:
Oh wait, is it "use the `default_results_per_page`, try to overwrite it, if that fails, it returns the default"?

Reply:
If `results_per_page` has either a value of `null` or a non-integer value in the config it will hit one of the two exceptions and return `results_per_page` which is still `default_results_per_page`

Reply:
yes, the second thing you said :)